src/components/MyEvents.js

```diff
@@ -171,6 +175,20 @@ const MyEvents = () => {
         setIsCancelDialogOpen(false);
     };
 
+    const [selectedEventParticipants, setSelectedEventParticipants] = React.useState([]);
+
+    const handleOpenParticipantsDialog = (event) => {
+        setSelectedEvent(event);
+        //console.log("current event is: " + event.id + " " + event.name);
+        setIsParticipantsListOpen(true);
+        setSelectedEventParticipants(participantsList.filter(x => x.id === event.id));
```

Comment:
ideally, you would filter through the participants in the API. Your query would be:
```
router.post('/api/getParticipantsByEvent', (req, res) => {
    let connection = mysql.createConnection(config);

    let sql =
        `SELECT events.id, CONCAT(users.firstName, " ", users.lastName) AS participantName
    FROM eventParticipants, events, users
    WHERE participantID = users.id
    AND eventID = events.id 
    AND eventID = ?
    ORDER BY events.id`;
    
    let data = [req.body.eventID];
    ...
```
    
instead of calling your `handleGetParticipants` on render, you would call it whenever the "See Participants" button is clicked. 

Doing the above is more consistent with what we have with the rest of the APIs, our APIs aren't filtering the response, only displaying it to the front-end. However, it does work with no issues. I'll leave it up to you whether or not you want to make this change.

Reply:
I originally did have it that way, but I figured it wasn't as efficient to keep having to call the api everytime you want to see a basic list. Though I see that its also not that efficient to have to filter. Which one is more inefficient?